### PR TITLE
plugin/kubernetes: Support both v1 and v1beta1 EndpointSlices

### PR DESCRIPTION
**Pull request by chrisohaver**
### 1. Why is this pull request needed and what does it do?

Adds support for both v1 and v1beta1 EndpointSlices.
 AFAIK, v1 and v1beta1 are functionally identical, however using v1beta1 on a K8s 1.21 cluster results in a deprecation notice warning in the log.  Using v1 on a K8s 1.21 cluster averts the warning. So, this code is essentially in place for now to remove a warning from the log. I welcome opinions as to whether or not it's worth adding this for that reason.

In this change, v1 will be used if available, if not, v1beta1 is used.

Like the dual Endpoint/EndpointSlice support, this code can be temporary until versions of K8s that do not support the v1 are no longer supported, at which point this code can be removed.

### 2. Which issues (if any) are related?

#4568

### CODE REVIEW

File: plugin/kubernetes/kubernetes.go
Comment (sgreene570) [inline plugin/kubernetes/kubernetes.go]: This comment should most likely be updated to reflect the new behavior of `endpointSliceSupported`.

File: plugin/kubernetes/kubernetes.go
Comment (sgreene570) [inline plugin/kubernetes/kubernetes.go:265]: Should this have a comment along the lines of:
"This can removed when all supported k8s versions no longer support EndpointSlice v1beta1" ?

Comment (chrisohaver) [reply plugin/kubernetes/kubernetes.go:265]: Actually, the entire function should be removed before that happens (or at the same time). IOW, once all supported k8s versions support v1, we can simply use v1, with no need for legacy endpoints or beta slices fallback.

I updated the function's comments to reflect that.

**Comment by sgreene570** [comment]
Friendly ping on this. Any updates here? :grin:

**Comment by chrisohaver** [comment]
> Friendly ping on this. Any updates here? 😁

No updates. Just waiting on a review from a maintainer.

**Review by johnbelamaric — APPROVED** [review]

**Comment by chrisohaver** [comment]
At first glance... it looks like merging this broke the unready endpoint fix ...

**Comment by chrisohaver** [comment]
> At first glance... it looks like merging this broke the unready endpoint fix ...

should be fixed with ... #4616

**Pull request merged**
